src/coreclr/src/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -1244,6 +1244,13 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
             //
 
             targetMethod = methodAfterConstraintResolution;
+            if (!_compilation.CompilationModuleGroup.VersionsWithMethodBody(targetMethod))
+            {
+                // This loosely corresponds to the bit around
+                // https://github.com/dotnet/runtime/blob/9380bbf37bb0ba19c7e2671359875c93686b400c/src/coreclr/src/vm/zapsig.cpp#L1278
+                // where we're resorting to use the original token for the sake of version resiliency.
+                targetMethod = originalMethod;
```

Comment:
Do we need to undo other effects of the constraint resolution? E.g. set `resolvedConstraint` back to false?

Reply:
Thinking about it some more it may be better to push this a bit up to the place where we handle the constraint resolution in general so that we only conditionally apply it when the resolved method belongs to the same version bubble. I'll fix that together with other PR feedback based on my attempts at clarifications below.